perf-tool/src/client.cpp

```diff
@@ -0,0 +1,184 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "client.hpp"
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+
+#include "utils.hpp"
+
+#define POLL_INTERVAL 150
+#define BUFFER_SIZE 4096
+
+using namespace std;
+
+Client::Client(int _portno, const string _hostname, bool _interactive_mode)
+{
+    portno = _portno;
+    hostname = _hostname;
+    interactive_mode = _interactive_mode;
+}
```

Comment:
Best practice to use member initializer lists: https://en.cppreference.com/w/cpp/language/constructor
```
Client::Client(int _portno, const string _hostname, bool _interactive_mode) : 
portno(_portno),
hostname(_hostname),
interactive_mode(_interactive_mode)
{}
```
good practice nitpick: member variables are the ones that normally start with underscore.